checkov/terraform/graph_builder/local_graph.py

```diff
@@ -274,20 +274,31 @@ def get_module_vertices_mapping(self) -> None:
         For each vertex, if it's originated in a module import, add to the vertex the index of the
         matching module vertex as 'source_module'
         """
+        module_lookup = {}
+        for module_idx in self.vertices_by_block_type[BlockType.MODULE]:
+            module_vertex = self.vertices[module_idx]
+            composed_key = (
+                module_vertex.name,
+                module_vertex.path,
+                module_vertex.source_module_object,
+                module_vertex.for_each_index,
+            )
+            module_lookup[composed_key] = module_idx
+
+        # Match vertices using the lookup
         for vertex in self.vertices:
-            if not vertex.source_module_object:
+            smo = vertex.source_module_object
```

Comment:
I don't think `smo` is readable enough, you can keep it as `source_module_object`